### DISCUSSION TITLE
Download a file without using "StaticFilesEndpoint"

### DISCUSSION THREAD

**Discussion by JadKHaddad** [poem-web/poem#264] [Q&A] [answered]
Hello. Is there a way to send a file from the poem-server to the browser without using "StaticFilesEndpoint". Say I want to respond to a post request with a file and not with text. Is it possible?

This functionality would be similar to python flask "send_from_directory"

**Comment by sunli829** [comment, marked as answer]
If I understand correctly, `send_from_directory` is used to send a single file?

Well, what you want should be this:

https://docs.rs/poem/1.3.29/poem/web/struct.StaticFileRequest.html

```rust
#[handler]
async fn index(req: StaticFileRequest) -> Result<impl IntoResponse> {
     Ok(req.create_response("/path/to/file", true)?)
}
````

**Reply by JadKHaddad** [reply]
Yes ```send_from_directory``` sends a single file.  

Thank you for your answer. It was exactly what I'm looking for.
